src/mdfour.c

```diff
@@ -228,7 +228,7 @@ inline static void rs_mdfour_block(rs_mdfour_t *md, void const *p)
    buffer first. */
 inline static void rs_mdfour_block(rs_mdfour_t *md, void const *p)
 {
-    unsigned long ptrval = (unsigned long)p;
+    unsigned long long ptrval = (unsigned long long)p;
```

Comment:
Casting to a long long can be a little expensive on platforms where addresses are smaller than that. The point of this was to figure out if the pointer p is aligned for a uint32_t. Copying it into another variable to do this is unnecessarily complicated. After reading the following;

https://stackoverflow.com/questions/15868313/why-cant-you-do-bitwise-operations-on-pointer-in-c-and-is-there-a-way-around-t
https://stackoverflow.com/questions/1845482/what-is-uintptr-t-data-type

I think a better answer is to remove this variable entirely and change the 'if' statement that uses it below to;

if ((uintptr_t)p & 3) {

My only concern is uintptr_t is "optional" in C99 (which is what we target as our minimum compiler requirement), and I have no idea if there are critical platforms that don't have it. If there are platforms without it, I suspect we should work around that with a mix of cmake and config.h.